e2e/testcases/otel_collector_test.go

```diff
@@ -0,0 +1,216 @@
+	nomostest.DeletePodByLabel(nt, "app", ocmetrics.OpenTelemetry, false)
+	if err := nt.WatchForAllSyncs(); err != nil {
+		nt.T.Fatal(err)
+	}
+	if err := watchForOtelCollectorStatusCurrent(nt); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Checking the otel-collector log contains failure...")
+	_, err = retry.Retry(60*time.Second, func() error {
+		return validateDeploymentLogHasFailure(nt, ocmetrics.OtelCollectorName, ocmetrics.MonitoringNamespace, GCMExportErrorCaption)
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+}
+
+func validateDeploymentLogHasFailure(nt *nomostest.NT, deployment, namespace, errorString string) error {
+	nt.T.Helper()
+
+	args := []string{"logs", fmt.Sprintf("deployment/%s", deployment), "-n", namespace}
+	cmd := fmt.Sprintf("kubectl %s", strings.Join(args, " "))
+	out, err := nt.Shell.Kubectl(args...)
+	if err != nil {
+		nt.T.Logf("failed to run %q: %v\n%s", cmd, err, out)
+		return err
+	}
+
+	entry := strings.Split(string(out), "\n")
+	for _, m := range entry {
+		if strings.Contains(m, errorString) {
+			return nil
+		}
+	}
+	return fmt.Errorf("error expected in the log of deployment %s, namespace %s but found none", deployment, namespace)
+}
+
+func validateDeploymentLogHasNoFailure(nt *nomostest.NT, deployment, namespace, errorString string) error {
+	nt.T.Helper()
+
+	args := []string{"logs", fmt.Sprintf("deployment/%s", deployment), "-n", namespace}
+	cmd := fmt.Sprintf("kubectl %s", strings.Join(args, " "))
+	out, err := nt.Shell.Kubectl(args...)
+	if err != nil {
+		nt.T.Logf("failed to run %q: %v\n%s", cmd, err, out)
+		return err
+	}
+
+	entry := strings.Split(string(out), "\n")
+	for _, m := range entry {
+		if strings.Contains(m, errorString) {
+			return fmt.Errorf("failure found in the log of deployment %s, namespace %s: %s", deployment, namespace, m)
+		}
+	}
+	return nil
+}
+
+// func validateDeploymentstatuscurrent
+func watchForOtelCollectorStatusCurrent(nt *nomostest.NT) error {
+	tg := taskgroup.New()
+	tg.Go(func() error {
+		return nt.Watcher.WatchForCurrentStatus(kinds.Deployment(), ocmetrics.OtelCollectorName, ocmetrics.MonitoringNamespace)
+	})
+	return tg.Wait()
+}
+
+// Create a new Monitoring service client using application default credentials
+func createGCMClient(ctx context.Context) (*monitoringv2.MetricClient, error) {
+	client, err := monitoringv2.NewMetricClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return client, nil
+}
+
+// Make a ListTimeSeries request of a specific metric to GCM and look for the
+// exact metric by metricType and clusterName in response.
+// Note: metricType in this context is the metric descriptor name, for example
+// "custom.googleapis.com/opencensus/config_sync/apply_operations_total".
+func listMetricInGCM(ctx context.Context, nt *nomostest.NT, client *monitoringv2.MetricClient, timeDiff time.Duration, metricType string) *monitoringv2.TimeSeriesIterator {
+	startTime := time.Now().UTC().Add(timeDiff)
```

Comment:
nit: might be slightly clearer if this just uses Sub with a positive value, maybe called "range"

Reply:
They use Sub() to get the duration between t-u.. not quite aligned with Add where a time of t+diff is returned. https://github.com/golang/go/blob/master/src/time/time.go#L893

Reply:
Perhaps we could just provide a start time as an argument instead of using a delta? (see my comment above)

Reply:
Done